components/calendar/data/data/macros.rs

```diff
@@ -1,4 +1,17 @@
 // @generated
+/// Declares a data provider. You can then use `impl_data_provider` or
+/// key specific macros, like `impl_core_helloworld_v1` to add implementations.
```

Comment:
Question: why did you make this _declare_ the data provider type, rather than just implement the trait?

Reply:
I thought we might want to control how the type is defined, but I just realised it needs to be a unit struct anyway.

Reply:
What necessitates that it is a unit struct?

Reply:
it currently is and people construct it with the unit constructor

Reply:
Seems like we should support things like this:

```rust
struct MyProvider(/* some custom fields */);
impl DataProvider<FooM> for MyProvider { ... }
icu_datetime_data::impl_datetime_symbols_v1!(MyProvider);
```

Reply:
Hmm I think this makes sense, given that the client needs to decide the definition because they will need to construct it. It's probably enough for future use cases to be able to add consts to the type.